src/ucp/proto/proto_common.c

```diff
@@ -191,9 +191,11 @@ ucp_proto_common_get_frag_size(const ucp_proto_common_init_params_t *params,
                            UCP_PROTO_COMMON_INIT_FLAG_REMOTE_ACCESS |
                            UCP_PROTO_COMMON_INIT_FLAG_SEND_ZCOPY) &&
         (context->config.ext.rma_zcopy_seg_size != UCS_MEMUNITS_AUTO)) {
-        if (*max_frag_p < context->config.ext.rma_zcopy_seg_size) {
-            ucs_warn("max tl fragment size (%lu) cannot be smaller than (%lu)",
-                     *max_frag_p, *min_frag_p);
+        if (context->config.ext.rma_zcopy_seg_size < *min_frag_p) {
+            ucs_warn("requested fragment size is smaller than minimum allowed "
+                     "(%lu), adjusting it",
+                     *min_frag_p);
+            *max_frag_p = *min_frag_p;
         } else {
```

Comment:
is not this branch redundant now?

Reply:
I don't think so, for example. if `context->config.ext.rma_zcopy_seg_size > max_frag_p` then you would not want to take the value of `context->config.ext.rma_zcopy_seg_size`

Reply:
then logic should be the same as with min value

Reply:
you mean:
`*max_frag_p = ucs_min(ucs_max(context->config.ext.rma_zcopy_seg_size, *min_frag_p), *max_frag_p);`?

Reply:
I mean:
```
if (a < min_val) {
    warn(...);
    a = min_val;
} else if (a > max_val) {
    warn(...);
    a = max_val;
}
```

Reply:
could you pls describe the logic you are implementing here? It seems like we think differently

Reply:
I added a new config var (`rma_zcopy_seg_size`) in order to limit seg size in case TL support very large seg size.
The reason for this new config is to simplify testing of reset_request scenraios

Reply:
I mean the logic in this particular if/else section, how do you adjust the parameter

Reply:
The logic is intended to force an upper bound to max_frag value. 
If I understand correctly you think it should also effect min_frag value. 
but it is only supposed to effect max_frag.

Reply:
then the code should be simplified, pls make it straight forward according what it's doing